Disable schema inference for unboxedOnly kernels

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #28186 Remove c10->ATen registration forwarding
* #28187 Remove use_c10_dispatcher: unboxed_only flags
* #28185 Remove codegen for use_c10_dispatcher: no
* #27978 Migrate remaining ops to the c10 dispatcher
* #28208 Move complex extension test to c10
* #28184 Fix codegen for out operators
* #28183 Allow migrating factory methods to c10
* #28182 Fix overload names
* #28181 Add unsupported types to schema type parser
* **#27977 Disable schema inference for unboxedOnly kernels**

The only remaining reason why we couldn't move some ops from globalATenDispatch to the c10 dispatcher was that schema inference didn't support some use cases.
But actually, we don't need schema inference for these ops. By disabling it, we can move the remaining ops from globalATenDispatch to the c10 dispatcher.

Differential Revision: [D17929696](https://our.internmc.facebook.com/intern/diff/D17929696/)

smessmer committed:
Disable schema inference for unboxedOnly kernels

The only remaining reason why we couldn't move some ops from globalATenDispatch to the c10 dispatcher was that schema inference didn't support some use cases.
But actually, we don't need schema inference for these ops. By disabling it, we can move the remaining ops from globalATenDispatch to the c10 dispatcher.

Differential Revision: [D17929696](https://our.internmc.facebook.com/intern/diff/D17929696/)

[ghstack-poisoned]

smessmer committed:
Update on "Disable schema inference for unboxedOnly kernels"


The only remaining reason why we couldn't move some ops from globalATenDispatch to the c10 dispatcher was that schema inference didn't support some use cases.
But actually, we don't need schema inference for these ops. By disabling it, we can move the remaining ops from globalATenDispatch to the c10 dispatcher.

Differential Revision: [D17929696](https://our.internmc.facebook.com/intern/diff/D17929696/)

[ghstack-poisoned]

ezyang commented:
Shouldn't there still be a TODO at these locations though? Or at least some explanation about the nullptr?

smessmer commented:
The TODO is directly above each of these functions, saying they should be deleted. I also added a comment explaining the nullptr now.

smessmer committed:
Update on "Disable schema inference for unboxedOnly kernels"


The only remaining reason why we couldn't move some ops from globalATenDispatch to the c10 dispatcher was that schema inference didn't support some use cases.
But actually, we don't need schema inference for these ops. By disabling it, we can move the remaining ops from globalATenDispatch to the c10 dispatcher.

Differential Revision: [D17929696](https://our.internmc.facebook.com/intern/diff/D17929696/)

[ghstack-poisoned]
